sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/SaveIntoDataSourceCommand.scala

```diff
@@ -44,8 +45,23 @@ case class SaveIntoDataSourceCommand(
   override def innerChildren: Seq[QueryPlan[_]] = Seq(query)
 
   override def run(sparkSession: SparkSession): Seq[Row] = {
-    val relation = dataSource.createRelation(
-      sparkSession.sqlContext, mode, options, Dataset.ofRows(sparkSession, query))
+    var relation: BaseRelation = null
```

Comment:
I didn't get why do you this trick. Why cannot you just assign the result of `try {} ..` to `val BaseRelation = `

Reply:
I think it's best done like this, because `LogicalRelation` will require `relation` to be a `BaseRelation`, while we don't know if try/catch will return something, throw an exception or just pass

I think we should catch only certain errors here, and not all possible exceptions